### ISSUE TITLE
Product variant optional attributes

### ISSUE THREAD

**Issue by piotrantosz** [saleor/saleor#5701]
### What I'm trying to achieve
Right now, if you want to create new product variant of X product type which have 3 attributes, for example:
color
size
something

you can't leave any variant attribute value empty.

### Describe a proposed solution

I think it's kinda useless, we implemented easy fix for that:
**AttributeAssignmentMixin** 
func **_validate_input** we removed:
```
        if is_variant:
            return cls._check_input_for_variant(cleaned_input, attribute_qs)
        else:
            return cls._check_input_for_product(cleaned_input, attribute_qs)
```

and left just 

```
     return cls._check_input_for_product(cleaned_input, attribute_qs)
```

and it works as charm, but it would be great to remove all is_variant useless logic then.

About frontend: the only change, is to make same thing with variant attributes as with product attributes, so that user can change value to None and then empty list is sent by graphql. (empty list, not None value, that's important tho. But it's just same as with product attribute update)

So basically with removing extra code, and unifying product+variant attributes we can achieve better functionality.

### Screenshots or mockups
https://i.imgur.com/2Iz759C.png
brak == none, just translation

**Comment by Boquete** [comment]
also - proposed solution above, checks if attribute is required, so it also works with required / non required attributes.

**Comment by maarcingebala** [comment]
Thanks for the feedback! I'm gathering all issues related to attributes and we'll try to address them when we'll be reiterating on attributes system implementation (which I hope is going to happen before the release of next version).
